tests/datasets/test_dataset.py

```diff
@@ -80,3 +80,32 @@ def process(self):
         saved_graphs = pickle.load(f)
         assert mx.array_equal(saved_graphs[0].node_features, data.node_features)
     shutil.rmtree(path)
+
+
+def test_dataset_transform():
+    data_list = [
+        GraphData(node_features=mx.ones((10, 5)), edge_labels=mx.ones((10, 1)))
+        for _ in range(3)
+    ]
+
+    def custom_transform(graph):
+        graph.feat = "test_transform"
+        return graph
+
+    path = os.path.join("/".join(__file__.split("/")[:-1]), ".tests/")
+    shutil.rmtree(path, ignore_errors=True)
+
+    class Dataset1(Dataset):
+        def __init__(self, transform):
+            super().__init__("test_dataset", path, transform=transform)
+
+        def download(self):
+            pass
+
+        def process(self):
+            self.graphs = data_list
+
+    dataset = Dataset1(custom_transform)
+
+    for i in range(len(dataset)):
+        assert dataset[i].feat == "test_transform"
```

Comment:
Can you delete the content of `path` after this?
We'll have to implement a fixture to automatically do that for all files created during testing now that we have multiple datasets, but we'll do that in a later PR.

Reply:
Added a line to remove the files at the path.